### ISSUE TITLE
`optimizePackageImports` is not working with `react-icons` in dev mode

### ISSUE THREAD

**Issue by FilipiBrabo** [vercel/next.js#70666]
### Link to the code that reproduces this issue

https://github.com/FilipiBrabo/nextjs-react-icons

### To Reproduce

1. Start the application with analyze-bundle enabled `ANALYZE=true npm run dev`
2. Go to `localhost:3000` the bundle files should open in your browser

### Current vs. Expected behavior

The current behavior is that all icons from a `react-icon` pack are being included in the final bundle in dev mode. 

From the [optimizePackageImports docs](https://nextjs.org/docs/pages/api-reference/next-config-js/optimizePackageImports) I expected that `react-icons/*` would be tree shaken both in dev and in prod build.

I've also included a `lucide` icon to see the difference, and noticed that `optimizePackageImports` is working as expected.

These are the bundles generated in dev mode of the application provided. You can see that all `fa`, `bs` and `ai` icons are being included in the final dev bundle.

Client bundle:
![client-bundle](https://github.com/user-attachments/assets/eec9041e-7f48-4e2c-9f99-95c4780d22fb)

Node.js bundle:
![nodejs-bundle](https://github.com/user-attachments/assets/7e552e26-9f75-482b-9418-9796ee5d35dc)

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #123-Ubuntu SMP Mon Jun 10 08:16:17 UTC 2024
  Available memory (MB): 32012
  Available CPU cores: 16
Binaries:
  Node: 18.20.4
  npm: 10.7.0
  Yarn: N/A
  pnpm: 8.15.8
Relevant Packages:
  next: 15.0.0-canary.174 // Latest available version is detected (15.0.0-canary.174).
  eslint-config-next: N/A
  react: 19.0.0-rc-2d16326d-20240930
  react-dom: 19.0.0-rc-2d16326d-20240930
  typescript: 5.3.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Developer Experience, Performance

### Which stage(s) are affected? (Select all that apply)

next dev (local)

### Additional context

I didn't dig too much deep into this, but these were the things that I've tried:

1. Use `modularizeImports` to replace `react-icons` with `@react-icons/all-files`. You can check it out in the `next.config.js` file.
```js
{
  reactStrictMode: true,
  modularizeImports: {
    "react-icons/fa": {
      transform: "@react-icons/all-files/fa/{{ member }}",
    },
    "react-icons/bs": {
      transform: "@react-icons/all-files/bs/{{ member }}",
    },
    "react-icons/ai": {
      transform: "@react-icons/all-files/ai/{{ member }}",
    },
  },
}
```
2. Specify `react-icons` lib in `optimizePackageImports` config 
```js
{
  reactStrictMode: true,
  experimental: {
    optimizePackageImports: [
      "react-icons/fa",
      "react-icons/bs",
      "react-icons/ai",
    ],
    // or
    optimizePackageImports: ["react-icons"],
  },
}
```

This really affects DX if you have too many icons from different packs being imported.
I'm happy to help in any way to fix this issue, just need a little guidance :sweat_smile:

**Comment by samcx** [comment]
@FilipiBrabo Thank you for submitting an issue!

We are taking a look at this— <picture data-single-emoji=":frog-eyes:" title=":frog-eyes:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/frog-eyes/558eb9b0fd306c97.png" alt=":frog-eyes:" align="absmiddle"></picture>

**Comment by FilipiBrabo** [comment]
> @FilipiBrabo Thank you for submitting an issue!
> 
> We are taking a look at this— <img alt=":frog-eyes:" width="20" height="auto" src="https://camo.githubusercontent.com/6211d269bb970f906bb8774af70c4da24680b7950847fd7a7a9f0e9f0b0f6076/68747470733a2f2f656d6f6a692e736c61636b2d656467652e636f6d2f5430434151303054552f66726f672d657965732f353538656239623066643330366339372e706e67">

Thanks, @samcx! Feel free to reach out if you need any help. I’m happy to assist however I can, and I’m curious to learn more about the bundling/optimizing process 😃

**Comment by stychu** [comment]
NextJs v14.2.23

Doesn't seem the issue is resolved. 
Our Compile times avg around 40 SEC which is insanity for development..... after I remove all imports to `react-icons` it goes down 3-5 sec which is more acceptable but still significant.

**Comment by noah-rss** [comment]
Bumping this issue up... my team is experiencing this issue with Next.js version 14.2.4. It is severely degrading our development server.

**Comment by FilipiBrabo** [comment]
For anyone dealing with this issue, the current workaround is to install `@react-icons/all-files` lib and import every single icon individually.
[See this](https://github.com/react-icons/react-icons/issues/593#issuecomment-1446098993) for how to install the latest version.

**Comment by finalight** [comment]
is this being fixed? or are we going to rely on the `all-files` variant? 

that variant don't seems to have autocomplete also

**Comment by jophy-ye** [comment]
> This really affects DX if you have too many icons from different packs being imported.

@FilipiBrabo Are you implying that the 2 options under "Additional Context" are _successful_ alternatives you've tried, even though it affects DX?

**Comment by FilipiBrabo** [comment]
> > This really affects DX if you have too many icons from different packs being imported.
> 
> [@FilipiBrabo](https://github.com/FilipiBrabo) Are you implying that the 2 options under "Additional Context" are _successful_ alternatives you've tried?

@jophy-ye Nope, those were things that I've tried but didn't fix the issue. Will update the description to make that clearer 😃
